core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala

```diff
@@ -53,6 +67,10 @@ private[spark] object TaskLocation {
   // confusion.  See  RFC 952 and RFC 1123 for information about the format of hostnames.
   val inMemoryLocationTag = "hdfs_cache_"
 
+  // Check if SPARK_LOCAL_HOSTNAME is set. If not set, host names for task locations will be
+  // resolved to IP addresses to fix the issue mentioned in SPARK-10149.
+  def sparkLocalHostname: Option[String] = sys.env.get("SPARK_LOCAL_HOSTNAME")
```

Comment:
I mean, do we need to fetch the SPARK_LOCAL_HOSTNAME every time we reference it?

just make it as a `val` is fine, I guess